### PR TITLE
CI update - .NET 7 Preview 2

### PR DESCRIPTION
**Pull request by gewarren**
cc @carlossanlop

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]
System.Runtime.InteropServices looks good

File: xml/Microsoft.Extensions.Options/ValidateOptionsResult.xml [hunk 1]
```diff
@@ -121,6 +121,32 @@
         <AssemblyVersion>6.0.0.0</AssemblyVersion>
         <AssemblyVersion>7.0.0.0</AssemblyVersion>
       </AssemblyInfo>
+      <Attributes>
+        <Attribute FrameworkAlternate="dotnet-plat-ext-7.0">
+          <AttributeName Language="C#">[System.Diagnostics.CodeAnalysis.MemberNotNullWhen(true, "Failures")]</AttributeName>
```
Comment (gewarren) [inline xml/Microsoft.Extensions.Options/ValidateOptionsResult.xml:126]: @huangmin-ms These annotations seem excessive - there are 3 for each annotated property. See https://github.com/dotnet/runtime/blob/c74a3fdf01d57a14218056475f952aae4b6f4e2e/src/libraries/Microsoft.Extensions.Options/src/ValidateOptionsResult.cs#L38-L40 for source code.

File: xml/System.IO.Compression/BrotliDecoder.xml [hunk 2]
```diff
@@ -71,19 +71,11 @@
...
-The return value can be as follows:
-- <xref:System.Buffers.OperationStatus.Done?displayProperty=nameWithType>: `source` was successfully and completely decompressed into `destination`.
-- <xref:System.Buffers.OperationStatus.DestinationTooSmall?displayProperty=nameWithType>: There is not enough space in `destination` to decompress `source`.
-- <xref:System.Buffers.OperationStatus.NeedMoreData?displayProperty=nameWithType>: The decompression action is partially done at least one more byte is required to complete the decompression task. This method should be called again with more input to decompress.
-- <xref:System.Buffers.OperationStatus.InvalidData?displayProperty=nameWithType>: The data in `source` is invalid and could not be decompressed.
-
-           ]]></format>
-        </remarks>
+        <remarks>The return value can be as follows:
+            - <see cref="F:System.Buffers.OperationStatus.Done" />: <paramref name="source" /> was successfully and completely decompressed into <paramref name="destination" />.
```
Comment (gewarren) [inline xml/System.IO.Compression/BrotliDecoder.xml:75]: @gewarren to check rendering.

File: xml/System.IO.Compression/BrotliEncoder.xml [hunk 3]
```diff
@@ -66,11 +66,8 @@
         <remarks>To be added.</remarks>
         <exception cref="T:System.ArgumentOutOfRangeException">
           <paramref name="quality" /> is not between the minimum value of 0 and the maximum value of 11.
-
--or-
-
-<paramref name="window" /> is not between the minimum value of 10 and the maximum value of 24.
-        </exception>
+            -or-
```
Comment (gewarren) [inline xml/System.IO.Compression/BrotliEncoder.xml:69]: Backport to ///.

File: xml/System.IO.Compression/BrotliStream.xml [hunk 4]
```diff
@@ -435,19 +435,9 @@
...
-
-This method disposes the Brotli stream by writing any changes to the backing store and closing the stream to release resources.   
-
-Calling `DisposeAsync` allows the resources used by the <xref:System.IO.Compression.BrotliStream> to be reallocated for other purposes. For more information, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).  
-
-          ]]></format>
-        </remarks>
+        <remarks>The `DisposeAsync` method lets you perform a resource-intensive dispose operation without blocking the main thread. This performance consideration is particularly important in a Windows 8.x Store app or desktop app where a time-consuming stream operation can block the UI thread and make your app appear as if it is not working. The async methods are used in conjunction with the <see langword="async" /> and <see langword="await" /> keywords in Visual Basic and C#.
```
Comment (gewarren) [inline xml/System.IO.Compression/BrotliStream.xml:438]: Check rendering.

File: xml/System.IO.Compression/BrotliStream.xml [hunk 5]
```diff
@@ -683,19 +665,9 @@ If the operation is canceled before it completes, the returned task contains the
...
-
-This method read a maximum of `buffer.Length` bytes from the current stream and store them in `buffer`. The current position within the Brotli stream is advanced by the number of bytes read; however, if an exception occurs, the current position within the Brotli stream remains unchanged. This method will block until at least one byte of data can be read, in the event that no data is available. `Read` returns 0 only when there is no more data in the stream and no more is expected (such as a closed socket or end of file). The method is free to return fewer bytes than requested even if the end of the stream has not been reached.
-
- Use <xref:System.IO.BinaryReader> for reading primitive data types.
-
-          ]]></format>
-        </remarks>
+        <remarks>Use the <see cref="P:System.IO.Compression.BrotliStream.CanRead" /> property to determine whether the current instance supports reading. Use the <see langword="System.IO.Compression.BrotliStream.ReadAsync" /> method to read asynchronously from the current stream.
```
Comment (gewarren) [inline xml/System.IO.Compression/BrotliStream.xml:668]: @gewarren to check.

File: xml/System.IO.Compression/BrotliStream.xml [hunk 6]
```diff
@@ -861,14 +813,14 @@ If the operation is canceled before it completes, the returned task contains the
...
+            Reads a byte from the stream and advances the position within the stream by one byte, or returns -1 if at the end of the stream.
+            </summary>
         <returns>The unsigned byte cast to an <see cref="T:System.Int32" />, or -1 if at the end of the stream.</returns>
         <remarks>To be added.</remarks>
         <exception cref="T:System.InvalidOperationException">Cannot perform read operations on a <see cref="T:System.IO.Compression.BrotliStream" /> constructed with <see cref="F:System.IO.Compression.CompressionMode.Compress" />.
-
--or-
-
-<see cref="P:System.IO.Compression.BrotliStream.BaseStream" /> returned more bytes than requested in read.</exception>
+            -or-
```
Comment (gewarren) [inline xml/System.IO.Compression/BrotliStream.xml:822]: Backport.

**Review by kouvel — COMMENTED** [review]
System.Threading LGTM

File: xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml [hunk 7]
```diff
@@ -22,15 +22,26 @@
...
   </AssemblyInfo>
+  <AssemblyInfo>
+    <AssemblyName>System.Security.Cryptography</AssemblyName>
+    <AssemblyVersion>7.0.0.0</AssemblyVersion>
+  </AssemblyInfo>
   <TypeForwardingChain>
     <TypeForwarding From="netstandard" FromVersion="2.1.0.0" To="System.Security.Cryptography.X509Certificates" ToVersion="5.0.0.0" FrameworkAlternate="net-5.0" />
     <TypeForwarding From="netstandard" FromVersion="2.1.0.0" To="System.Security.Cryptography.X509Certificates" ToVersion="6.0.0.0" FrameworkAlternate="net-6.0" />
-    <TypeForwarding From="netstandard" FromVersion="2.1.0.0" To="System.Security.Cryptography.X509Certificates" ToVersion="7.0.0.0" FrameworkAlternate="net-7.0" />
+    <TypeForwarding From="netstandard" FromVersion="2.1.0.0" To="System.Security.Cryptography" ToVersion="7.0.0.0" FrameworkAlternate="net-7.0" />
```
Comment (gewarren) [inline xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml:32]: @carlossanlop Did this type move? Or did I set up the assemblies incorrectly?

File: xml/System.Web.DynamicData/DynamicFilter.xml [hunk 8]
```diff
@@ -417,9 +417,9 @@
         <Parameter Name="source" Type="System.Linq.IQueryable" Index="0" FrameworkAlternate="netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8" />
       </Parameters>
       <Docs>
-        <param name="source">The data source query.</param>
-        <summary>Returns the filter repeater.</summary>
-        <returns>The filter.</returns>
+        <param name="source">To be added.</param>
```
Comment (gewarren) [inline xml/System.Web.DynamicData/DynamicFilter.xml:420]: Losing doc comments here.

File: xml/System.Web.DynamicData/QueryableFilterRepeater.xml [hunk 9]
```diff
@@ -310,9 +310,9 @@
         <Parameter Name="source" Type="System.Linq.IQueryable" Index="0" FrameworkAlternate="netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8" />
       </Parameters>
       <Docs>
-        <param name="source">The data source query.</param>
-        <summary>Returns the filter repeater.</summary>
-        <returns>The filter.</returns>
+        <param name="source">To be added.</param>
```
Comment (gewarren) [inline xml/System.Web.DynamicData/QueryableFilterRepeater.xml:313]: Loss.

**Comment by gewarren** [comment]
Going to start from scratch.

**Pull request closed without merging**